src/atmospheric_thermodynamics.jl

```diff
@@ -182,9 +184,14 @@ function AtmosphereThermodynamics(FT = Oceananigans.defaults.FloatType;
     vapor = IdealGas(FT; molar_mass = vapor_molar_mass,
                          heat_capacity = vapor_heat_capacity)
 
-    return AtmosphereThermodynamics(convert(FT, molar_gas_constant),
-                                    convert(FT, gravitational_acceleration),
-                                    dry_air, vapor, saturation, condensation, deposition)
+    S = typeof(saturation)
+    C = typeof(condensation)
+    F = typeof(deposition)
+
+    return AtmosphereThermodynamics{FT, S, C, F}(convert(FT, molar_gas_constant),
+                                                 convert(FT, gravitational_acceleration),
+                                                 dry_air, vapor,
+                                                 saturation, condensation, deposition)
```

Comment:
I think it's better to infer the types (the code is identical, but easier to maintain and understand)

Reply:
There is no additional functionality added by explicitly writing `FT, S, C, F` --- at least I can't see that. We only need to write those types if 1) they cannot be inferred or 2) we want to convert. However, the code here _already_ calls `convert`.

Reply:
let me try removing it and see if things still work -- it might have been during my attempts to find why some types were note inferred and were Any